core/src/main/scala/org/apache/spark/scheduler/cluster/StandaloneSchedulerBackend.scala

```diff
@@ -153,7 +153,7 @@ private[spark] class StandaloneSchedulerBackend(
   override def executorRemoved(fullId: String, message: String, exitStatus: Option[Int]) {
     val reason: ExecutorLossReason = exitStatus match {
       case Some(code) => ExecutorExited(code, exitCausedByApp = true, message)
-      case None => SlaveLost(message)
+      case None => SlaveLost(message, workerLost = true /* worker loss event from master */)
```

Comment:
Went with propagating just `workerLost` explicitly all the way from the master, since ExecutorState is private to deploy.